spec/models/external_service/survey_response_spec.rb

```diff
@@ -83,6 +83,24 @@
       receiver = survey_response.save!
       expect(receiver.receiver.quantity).to eq quantity
     end
+
+    it "locks the quantity for order if there is a quantity param" do
+      params[:quantity] = order_detail.quantity
+      receiver = survey_response.save!
+      expect(order_detail.reload).to be_quantity_locked_by_survey
```

Comment:
Minor, but I'd probably have written these as `expect { survey_response.save! }.to change…` just to be sure the bit flipped. Also, is there any reason to assign `receiver`?